### PR TITLE
feat: structure for cardinality testing

### PR DESCRIPTION
**Pull request by patrickhyw**
**Summary**: Defined the main interfaces (`CardtestRunner` and `CardtestRunnerDBHelper`) and structure (the `optd-perftest` package) for cardinality testing. Wrote function to compute the Q-error of multiple databases.

### CODE REVIEW

File: optd-perftest/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+mod cardtest;
+mod postgres;
+
+fn main() {
+    let databases: Vec<Box<dyn CardtestRunnerDBHelper>> = vec![
+        Box::new(block_on(PostgresDb::new()).unwrap()),
+        Box::new(block_on(DatafusionDb::new()).unwrap()),
+    ];
+    let cardtest_runner = block_on(CardtestRunner::new(databases)).unwrap();
+    let qerrors = block_on(cardtest_runner.eval_qerrors("")).unwrap();
```
Comment (Gun9niR) [inline optd-perftest/src/main.rs:16]: If you think writing `block_on` everywhere is too cumbersome, you can use `#[tokio::main]` to change main into an async function.

File: optd-perftest/src/cardtest.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    async fn eval_true_card(&self, _sql: &str) -> anyhow::Result<usize> {
+        Ok(10)
+    }
+
+    async fn eval_est_card(&self, _sql: &str) -> anyhow::Result<usize> {
+        Ok(5)
+    }
+}
```
Comment (Gun9niR) [inline optd-perftest/src/cardtest.rs]: Might be better to move trait implementations to separate files.

**Review by Gun9niR — APPROVED** [review]
Rest LGTM!

File: optd-perftest/src/main.rs (hunk 1 above)
Comment (patrickhyw) [reply optd-perftest/src/main.rs:16]: thanks for the note. I changed this in the tpch-kit PR so I won't change it here

File: optd-perftest/src/cardtest.rs (hunk 2 above)
Comment (patrickhyw) [reply optd-perftest/src/cardtest.rs]: fixed

**Pull request merged**